api/v1alpha1/workspace_validation.go

```diff
@@ -45,7 +45,7 @@ func (w *Workspace) Validate(ctx context.Context) (errs *apis.FieldError) {
 		errs = errs.Also(w.validateCreate().ViaField("spec"))
 		if w.Inference != nil {
 			// TODO: Add Adapter Spec Validation - Including DataSource Validation for Adapter
-			errs = errs.Also(w.Resource.validateCreate(*w.Inference).ViaField("resource"),
+			errs = errs.Also(w.Resource.validateCreate(w.Inference, w.Tuning).ViaField("resource"),
 				w.Inference.validateCreate().ViaField("inference"))
 		}
 		if w.Tuning != nil {
```

Comment:
We need to have separate resource check for tuning.

Reply:
Added